src/tv2-common/__tests__/onTimelineGenerate.spec.ts

```diff
@@ -0,0 +1,348 @@
+			const resolvedPieces: IBlueprintResolvedPieceInstance[] = [
+				createPieceInstance('previousPiece', 0, 10, true, true),
+				createPieceInstance('currentPiece', 10, undefined, true, false)
+			]
+
+			const result = createSisyfosPersistedLevelsTimelineObject(resolvedPieces, LAYERS_THAT_WANTS_TO_BE_PERSISTED_ARRAY)
+
+			expect(result.content.channels).toHaveLength(1)
+		})
+
+		it('should not persist anything when current piece dont accept and dont want to persist', () => {
+			const resolvedPieces: IBlueprintResolvedPieceInstance[] = [
+				createPieceInstance('previousPiece', 0, 10, true, true),
+				createPieceInstance('currentPiece', 10, undefined, false, false)
+			]
+
+			const result = createSisyfosPersistedLevelsTimelineObject(resolvedPieces, LAYERS_THAT_WANTS_TO_BE_PERSISTED_ARRAY)
+
+			expect(result.content.channels).toHaveLength(0)
+		})
+
+		it('should persist when previous piece does not accept persist, but current does accept', () => {
+			const resolvedPieces: IBlueprintResolvedPieceInstance[] = [
+				createPieceInstance('previousPiece', 0, 10, true, false),
+				createPieceInstance('currentPiece', 10, undefined, false, true)
+			]
+
+			const result = createSisyfosPersistedLevelsTimelineObject(resolvedPieces, LAYERS_THAT_WANTS_TO_BE_PERSISTED_ARRAY)
+
+			expect(result.content.channels).toHaveLength(1)
+		})
+
+		it('should persist when current piece accepts persist and duration is not undefined', () => {
+			const resolvedPieces: IBlueprintResolvedPieceInstance[] = [createPieceInstance('currentPiece', 0, 5, false, true)]
+
+			const result = createSisyfosPersistedLevelsTimelineObject(resolvedPieces, LAYERS_THAT_WANTS_TO_BE_PERSISTED_ARRAY)
+
+			expect(result.content.channels).toHaveLength(1)
+		})
+
+		it('should persist all previous layers that wants to be persisted', () => {
+			const firstLayerThatWantToBePersisted: string = 'firstLayer'
+			const secondLayerThatWantToBePersisted: string = 'secondLayer'
+			const thirdLayerThatWantToBePersisted: string = 'thirdLayer'
+			const layersThatWantToBePersisted: SisyfosPersistMetaData['sisyfosLayers'] = [
+				firstLayerThatWantToBePersisted,
+				secondLayerThatWantToBePersisted,
+				thirdLayerThatWantToBePersisted
+			]
+			const resolvedPieces: IBlueprintResolvedPieceInstance[] = [createPieceInstance('currentPiece', 0, 5, true, true)]
+
+			const result = createSisyfosPersistedLevelsTimelineObject(resolvedPieces, layersThatWantToBePersisted)
+
+			expect(
+				result.content.channels.some(channel => channel.mappedLayer === firstLayerThatWantToBePersisted)
+			).toBeTruthy()
+			expect(
+				result.content.channels.some(channel => channel.mappedLayer === secondLayerThatWantToBePersisted)
+			).toBeTruthy()
+			expect(
+				result.content.channels.some(channel => channel.mappedLayer === thirdLayerThatWantToBePersisted)
+			).toBeTruthy()
+		})
+
+		it('cuts to executeAction that dont accept persist, dont persist layers', () => {
```

Comment:
But it doesn't replace a primary piece?
If we are on a LIVE and use the executeAction for cutToCamera, that executeAction will add a new Piece for the camera to the Part, so we now have a Part with two Pieces, LIVE and CAM, which is the scenario in this test.

If we already had cut to a camera and then cut to another cam, then yes, it would replace the piece, but that's not the scenario of the test.

Reply:
Stop one piece, start another one. That's replacement to me. The CAM->CAM scenario is an update, not a replacement, because the id is still the same.